src/utils/nextPurchaseDay.js

```diff
@@ -0,0 +1,25 @@
+/***
+ * Firts we verified by specificity and
+ * we return the string that is used for the
+ * aria-label and to add the proper style
+ */
+
+export const nextPurchaseDay = (
+  daysUntilNextPurchase,
+  lastPurchaseDate,
+  numberOfPurchases,
+) => {
+  const nextEstimatedPurchase = 2 * daysUntilNextPurchase;
+  if (numberOfPurchases === 1 || lastPurchaseDate === nextEstimatedPurchase) {
```

Comment:
I think that this way we can make a more accurate comparison. What do you think?

```suggestion
  if (numberOfPurchases === 1 || lastPurchaseDate >= nextEstimatedPurchase) {
```

Reply:
When I try to do what you say all logic disappears
![image](https://user-images.githubusercontent.com/42523734/141367175-a32afebf-a33a-46a3-a0ff-235927559c23.png)

Reply:
Yes, that is what happens when we use ` if (numberOfPurchases === 1 || lastPurchaseDate >= nextEstimatedPurchase) ` :cry:

Reply:
I did some tests to try to better explain my suggestion/question. As I understand it, the product is considered inactive when the last purchase was made at least twice as long (in days) as what would be considered "ideal". For example: I have a product in my list that should be purchased again in 5 days, but my last purchase was made 10 days ago. In this scenario, this product would be considered inactive because my last purchase is twice as long (in days) than what would be ideal. In a second example, I have a product that was supposed to be purchased in 5 days, but my last purchase was 15 days ago. This is more than double, but it should also be considered "inactive". So I suggested that the comparison be done using "<=". From your tests, I was curious to know why the comparison was not working, and I decided to test it myself. I ended up noticing that the comparison is being done between different data types. `nextEstimatedPurchase` is calculated in days, while the comparison with `lastPurchaseDate` is done with the type "Date". I think the problem is in this case. The ideal would be to compare two dates. We can think of a solution together, but at this first moment I think we could calculate the maximum purchase term (2 * daysUntilNextPurchase), turn it into a date and thus compare the difference in days between `lastPurchaseDate` and `nextEstimatedPurchase`. To try to illustrate better I created a test in Firebase, it's in `isa-collection`. In it I have a product that needs to be purchased within 5 days, but your last purchase was a month ago. I believe in this scenario it should be "inactive", but it still appears as "active" in the list. We can confirm with mentors that the understanding is correct. If I'm getting it all wrong, please let me know! You guys did a great job, but I am really confused by the requirements of this issue 😢

![image](https://user-images.githubusercontent.com/32148199/141389494-f06c0cc9-2994-4dd1-be9e-17492331a79f.png)

![image](https://user-images.githubusercontent.com/32148199/141389551-76dccfa7-776f-4121-8314-9dc875af5203.png)

Reply:
Ok, this got me thinking :thinking: :thinking:  and you're right. I went through the notes:

> - El criterio para que un artículo esté inactivo ha confundido a los equipos en el pasado. Se ha reformulado para que sea más claro, pero lo esencial es que un artículo está inactivo si el tiempo transcurrido desde su última compra es el doble (o más) de lo estimado la última vez que se compró.
> - Por ejemplo, si la última estimación fue de 14 días y han transcurrido 28 días, el artículo se considera inactivo.

As it says ***tiempo transcurrido desde su última compra**  es el doble ( o más ) de lo **estimado la ultima vez que se compró*** made me realize we need to transform the `lastPurchaseDate` to days (using today as a reference to get the days) and then compare it with >= `daysUntilNextPurchase` field
![image](https://user-images.githubusercontent.com/37032132/141409961-107bcb82-df4e-439a-9145-78a786e88c6b.png)

Reply:
Thank you @yanurd and @susird! I just tested it and now it's working perfectly! ✨